features/include/pcl/features/good.h

```diff
@@ -0,0 +1,326 @@
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the copyright holder(s) nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *
+ *
+ */
+
+
+#ifndef PCL_FEATURES_GOOD_H_
+#define PCL_FEATURES_GOOD_H_
+
+#include <pcl/common/transforms.h>
+#include <pcl/common/common.h>
+
+namespace pcl
+{
+/** \brief GOOD: a Global Orthographic Object Descriptor for 3D object recognition and manipulation.
+  * GOOD descriptor has been designed to be robust, descriptive and efficient to compute and use. 
+  * It has two outstanding characteristics: 
+  * 
+  * (1) Providing a good trade-off among :
+  *	- descriptiveness,
+  *	- robustness,
+  *	- computation time,
+  *	- memory usage.
+  * 
+  * (2) Allowing concurrent object recognition and pose estimation for manipulation.
+  * 
+  * \note This is an implementation of the GOOD descriptor which has been presented in the following papers:
+  * 
+  *	[1] Kasaei, S. Hamidreza,  Ana Maria Tomé, Luís Seabra Lopes, Miguel Oliveira 
+  *	"GOOD: A global orthographic object descriptor for 3D object recognition and manipulation." 
+  *	Pattern Recognition Letters 83 (2016): 312-320.http://dx.doi.org/10.1016/j.patrec.2016.07.006
+  *
+  *	[2] Kasaei, S. Hamidreza, Luís Seabra Lopes, Ana Maria Tomé, Miguel Oliveira 
+  * 	"An orthographic descriptor for 3D object learning and recognition." 
+  *	2016 IEEE/RSJ International Conference on Intelligent Robots and Systems (IROS), Daejeon, 2016, 
+  *	pp. 4158-4163. doi: 10.1109/IROS.2016.7759612
+  * 
+  * Please adequately refer to this work any time this code is being used by citing above papers.
+  * If you do publish a paper where GOOD descriptor helped your research, we encourage you to cite the above papers in your publications.
+  * 
+  * \author Hamidreza Kasaei (Seyed.Hamidreza[at]ua[dot]pt)
+  */
+  
+  template <typename PointInT>
+  class GOODEstimation
```

Comment:
@SergioRAgostinho @taketwo: there is a point that puzzles me. 
I try to change it as : 

     class PCL_EXPORTS GOODEstimation :  public Feature<PointInT, PointOutT>

In the case of the GOOD descriptor, the length of shape descriptor depends on the number of bins, n, which is calculated as 3 * n^2.

I could not find a proper way to define PointOutT as an std::vecotr<float>; 
Although I see that the other descriptors defined arrays with pre-defined size, I prefer to have it dynamic.

Any suggestion is highly welcomed.